src/bc_common.d

```diff
@@ -102,12 +102,18 @@ enum BCTypeEnum : ubyte
 
 }
 
+enum BCTypeFlags
+{
+  Const = 0x1,
+}
 struct BCType
 {
     BCTypeEnum type;
     alias type this;
     /// 0 means basic type
     uint typeIndex;
+    // additional information
+    uint flags;
```

Comment:
A rather annoying test, just to support a bit of casting const away.
Maybe actually modifying the value might be a saner choice than adding a runtime const check.